### PR TITLE
[SPARK-37928][SQL][TESTS] Add Parquet Data Page V2 test scenario to `DataSourceReadBenchmark`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This PR adds a corresponding `Parquet Data Page V2` test scenario for each `Parquet Data Page V1` test scenario to `DataSourceReadBenchmark`.

### Why are the changes needed?
Add micro benchmark for `Parquet Data Page V2`.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GA

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @dongjoon-hyun @sunchao @viirya I give this pr to adds a corresponding `Parquet Data Page V2` test scenario for each `Parquet Data Page V1` test scenario to `DataSourceReadBenchmark`,  if @parthchandra plans to do this part of work in SPARK-37912, I will close this pr.

**Comment by parthchandra** [comment]
I was planning to add these in subsequent PR)s) which includes implementation of the remaining encodings for Parquet V2. 
But since this is done, and supersedes my PR, let's merge this one and I can close my PR (or rename it to just address the review comments for the original delta binary PR).

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DataSourceReadBenchmark.scala [hunk 1]
```diff
@@ -167,87 +172,103 @@ object DataSourceReadBenchmark extends SqlBasedBenchmark {
...
-            case FloatType => (col: InternalRow) => doubleSum += col.getFloat(0)
-            case DoubleType => (col: InternalRow) => doubleSum += col.getDouble(0)
-          }
+        withParquetVersions { version =>
+          // Driving the parquet reader in batch mode directly.
+          val files = TestUtils.listDirectory(new File(dir, s"parquet$version"))
+          val enableOffHeapColumnVector = spark.sessionState.conf.offHeapColumnVectorEnabled
+          val vectorizedReaderBatchSize = spark.sessionState.conf.parquetVectorizedReaderBatchSize
+          // Decoding in vectorized but having the reader return rows.
+          parquetReaderBenchmark
```
Comment (sunchao) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DataSourceReadBenchmark.scala:228]: nit: I think we can just put this in the first `withParquetVersions` above?

**Review by sunchao — APPROVED** [review]
LGTM

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DataSourceReadBenchmark.scala (hunk 1 above)
Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DataSourceReadBenchmark.scala:226]: 4611bbd fix this

**Comment by LuciferYang** [comment]
> I was planning to add these in subsequent PR)s) which includes implementation of the remaining encodings for Parquet V2.
> But since this is done, and supersedes my PR, let's merge this one and I can close my PR (or rename it to just address the review comments for the original delta binary PR).

thanks ~ @parthchandra

**Comment by sunchao** [comment]
Thanks @LuciferYang , merged to master!

**Comment by LuciferYang** [comment]
thanks all

**Comment by dongjoon-hyun** [comment]
+1, late LGTM. Thank you all!

**Pull request closed without merging**